velox/common/memory/MappedMemory.cpp

```diff
@@ -59,17 +60,57 @@ void MappedMemory::Allocation::findRun(
   VELOX_CHECK(false, "Seeking to an out of range offset in Allocation");
 }
 
-// Rounds 'value' to the next multiple of 'factor'.
-template <typename T, typename U>
-static inline T roundUp(T value, U factor) {
-  return (value + (factor - 1)) / factor * factor;
+MachinePageCount MappedMemory::ContiguousAllocation::numPages() const {
+  return bits::roundUp(size_, kPageSize) / kPageSize;
 }
 
 // static
 void MappedMemory::destroyTestOnly() {
   instance_ = nullptr;
 }
 
+std::string MappedMemory::toString() const {
+  return fmt::format("MappedMemory: Allocated pages {}", numAllocated());
+}
+
+MachinePageCount MappedMemory::allocationSize(
+    MachinePageCount numPages,
+    MachinePageCount minSizeClass,
+    std::array<int32_t, kMaxSizeClasses>& sizeIndices,
+    std::array<int32_t, kMaxSizeClasses>& sizeCounts,
+    int32_t& numSizes) const {
+  int32_t needed = numPages;
+  int32_t pagesToAlloc = 0;
+  numSizes = 0;
+  VELOX_CHECK_LE(
+      minSizeClass,
+      sizes_.back(),
+      "Requesting minimum size larger than largest size class");
```

Comment:
It would be helpful or troubleshooting to add requested size class and largest available in the error message.

Reply:
Repeating in case this comment got lost: It would be helpful or troubleshooting to add requested size class and largest available in the error message.